### PR TITLE
Add tests for Git remote URL changes

### PR DESCRIPTION
**Pull request by rcutmore**
##### ISSUE TYPE

- Bugfix Pull Request
##### ANSIBLE VERSION

```
ansible 2.2.0 (git-integration-tests 70e58ffcd8) last updated 2016/07/30 11:26:19 (GMT -400)
  lib/ansible/modules/core: (issue-4006 4f493ff7ca) last updated 2016/07/31 13:31:41 (GMT -400)
  lib/ansible/modules/extras: (detached HEAD 68ca157f3b) last updated 2016/07/18 07:22:17 (GMT -400)
  config file = 
  configured module search path = Default w/o overrides

```
##### SUMMARY

This updates existing Git integration tests and adds new tests for changes to the Git module ([PR here](https://github.com/ansible/ansible-modules-core/pull/4243)). Both pull requests should be merged together.

### CODE REVIEW

File: test/integration/roles/test_git/tasks/main.yml [hunk 1]
```diff
@@ -435,6 +437,60 @@
...
+  git: repo={{ repo_update_url_1 }} dest={{ checkout_dir }}
+  register: checkout_same_url
+
+- name: check repo not changed (when get url command supported in git version)
+  assert:
+    that:
+      - not checkout_same_url|changed
+  when: git_version.stdout | version_compare("{{git_version_supporting_get_url}}", '>=')
+
+- name: check repo changed (when get url command not supported in git version)
```
Comment (robinro) [inline test/integration/roles/test_git/tasks/main.yml]: I would not ensure that the bug is there. Only test for desired behavior. Otherwise someone fixing this behavior in the future will make these tests fail.

File: test/integration/roles/test_git/tasks/main.yml [hunk 2]
```diff
@@ -435,6 +437,60 @@
...
+- name: check repo changed
+  assert:
+    that:
+      - checkout_new_url|changed
+
+
+- name: clone repo with new url in check mode
+  git: repo={{ repo_update_url_1 }} dest={{ checkout_dir }}
+  register: checkout_new_url_check_mode
+  check_mode: True
```
Comment (robinro) [inline test/integration/roles/test_git/tasks/main.yml]: Yeah check_mode!

File: test/integration/roles/test_git/tasks/main.yml [hunk 3]
```diff
@@ -495,7 +551,7 @@
 
 - assert:
     that: "not checkout2|changed"
-  when: git_version.stdout | version_compare("{{git_version_supporting_depth}}", '>=')
+  when: git_version.stdout | version_compare("{{git_version_supporting_get_url}}", '>=')
```
Comment (robinro) [inline test/integration/roles/test_git/tasks/main.yml]: Why did you change this? Looks like too much search/replace.

File: test/integration/roles/test_git/tasks/main.yml [hunk 4]
```diff
@@ -555,14 +611,19 @@
   git: repo={{ checkout_dir }} dest={{checkout_dir}}.copy version=newtag
   register: update_new_tag
 
+- name: check new head not changed
+  assert:
+    that:
+      - not update_new_tag|changed
+  when: git_version.stdout | version_compare("{{git_version_supporting_get_url}}", '>=')
```
Comment (robinro) [inline test/integration/roles/test_git/tasks/main.yml]: why do you check for version here?

File: test/integration/roles/test_git/tasks/main.yml (hunk 3 above)
Comment (rcutmore) [inline test/integration/roles/test_git/tasks/main.yml]: The tests that check for the version fail without the version check. Versions lower than 2.7 will report being changed due to not having `git remote get-url`.

File: test/integration/roles/test_git/tasks/main.yml (hunk 1 above)
Comment (rcutmore) [inline test/integration/roles/test_git/tasks/main.yml]: Got it, will do.

**Comment by rcutmore** [comment]
@robinro Thanks for reviewing. I update these tests along with the change to use `git ls-remote --get-url`.

File: test/integration/roles/test_git/tasks/main.yml (hunk 3 above)
Comment (robinro) [inline test/integration/roles/test_git/tasks/main.yml]: So for this you need version >-= get_url and version >= depth, i.e. one can revert it, once you go to ls-remote (since 1.9 for depth is bigger than 1.7 for ls-remote)

**Comment by rcutmore** [comment]
@robinro I updated the tests and ran them against a newer and older version of Git. From what I could see `ls-remote --get-url` was added in 1.7.5, but I can update this if I'm mistaken.

File: test/integration/roles/test_git/tasks/main.yml [hunk 5]
```diff
@@ -495,7 +545,7 @@
 
 - assert:
     that: "not checkout2|changed"
-  when: git_version.stdout | version_compare("{{git_version_supporting_depth}}", '>=')
+  when: git_version.stdout | version_compare("{{git_version_supporting_depth}}", '>=') and git_version.stdout | version_compare("{{git_version_supporting_depth}}", '>=')
```
Comment (robinro) [inline test/integration/roles/test_git/tasks/main.yml]: the second check should be version_geturl

Comment (rcutmore) [inline test/integration/roles/test_git/tasks/main.yml]: Thanks for catching that, just pushed a fix.

**Comment by robinro** [comment]
@jctanner Please consider merging this. You already merged the changes to the git module https://github.com/ansible/ansible-modules-core/pull/4243, which are covered by these tests.

**Comment by jctanner** [comment]
@robinro @rcutmore very sorry I missed this. Assigning to myself.

**Comment by jctanner** [comment]
@rcutmore github thinks there is a merge conflict although I don't see one locally. Can you rebase and maybe add a new commit so we can try to trigger shippable to retest and github to re-check?

**Comment by mattclay** [comment]
The integration test roles have moved, which is why this PR needs to be updated.

**Comment by rcutmore** [comment]
@jctanner I rebased this PR but more recent changes have broken the check_mode test when running them locally. It looks like it has something to do with [this commit](https://github.com/ansible/ansible-modules-core/commit/e4f40d1cdda9161849098bfeac9d360fe9dd77f3) because the test passes before that point in time but I haven't had a chance to investigate further yet.

**Comment by rcutmore** [comment]
@jctanner I updated this PR for the recent repo merge and added the required fix that was submitted to the old core modules repo before the merge ([here](https://github.com/ansible/ansible-modules-core/pull/5374)).

**Comment by rcutmore** [comment]
ready_for_review

**Comment by mattclay** [comment]
@rcutmore Thanks for the fix and the new tests!
